src/app/patients/results/observation-selector.directive.js

```diff
@@ -381,7 +381,7 @@ function observationSelector(store) {
           });
 
           // Get's patients groups sorted by name
-          scope.groups = _.sortBy(scope.patient.getGroups(), 'shortName');
+          scope.groups = _.sortBy(scope.patient.getGroups(), 'weight');
```

Comment:
This change sorts all observations by weight (if weight is provided), including in the all tab. However, in the all tab, observations should be sorted by name alphabetically by default.